presto-parser/src/main/antlr4/io/prestosql/sql/parser/SqlBase.g4

```diff
@@ -42,6 +42,7 @@ statement
         (WITH properties)?                                             #createSchema
     | DROP SCHEMA (IF EXISTS)? qualifiedName (CASCADE | RESTRICT)?     #dropSchema
     | ALTER SCHEMA qualifiedName RENAME TO identifier                  #renameSchema
+    | ALTER SCHEMA qualifiedName SET AUTHORIZATION principal           #setSchemaAuthorization
```

Comment:
Per the SQL spec, the authorization identifier should follow this syntax:

```
<authorization identifier> ::=
    <role name>
  | <user identifier>

<role name> ::=
  <identifier>

<user identifier> ::=
  <identifier>
```

`principal` in this grammar file is defined as:

```
principal
    : identifier            #unspecifiedPrincipal
    | USER identifier       #userPrincipal
    | ROLE identifier       #rolePrincipal
    ;
```

Which doesn't fit what the spec describes for this case.

Reply:
We may want to extend it in the future, but I'd prefer we start conservatively and follow the spec as closely as possible.

Reply:
I thought it was nice that the principal parsing would be the same across GRANT, REVOKE, and SET AUTHORIZATION.

Do we stay closer to the SQL standard or to Presto's existing syntax in similar scenarios?

Also in the standard you cannot set authorization for a user if the user happens to have the same name as a role (and for some database - such as PostgreSQL - where a user and a role are the same thing that's fine).

In order to know whether you set to a user or a role you'd need to check whether such a role does exist.

Happy to change, I just think it's better this way.

Reply:
Please let me know.

BTW. I looked up the standard system for GRANT and REVOKE and they also use what you cite above:
```
<authorization identifier>    ::=   <role name> | <user identifier>

<role name>    ::=   <identifier>

<user identifier>    ::=   <identifier>
```

Given that we already diverged from the SQL standard with GRANT and REVOKE I'd vote for being consistent within Presto.

Also one last point: Unless we force the explicit ROLE syntax, since Presto and Hive do not have pre-defined users, there would be no way to enforce valid roles, when a role is misspelled it would just need to assume that you meant a user.

In the end... Let me know what's the best approach and I'll do that.

Reply:
> Given that we already diverged from the SQL standard with GRANT and REVOKE I'd vote for being consistent within Presto.

Yeah, it's unfortunate we diverged from the standard like that. Let me think about it and I'll get back to you shortly.

Reply:
Awesome. Thanks!

Reply:
I apologize for the delay, I should've responded sooner. Let's keep it as you have it for now. It covers the standard case plus the presto-specific extension. We can revisit later more holistically if we decide it's worth constraining.

Reply:
Thanks for looking @martint ! (I do realize you have many, many PRs to look through.)
For this one then I'm assuming that are currently no further requests for changes.